service/history/events/cache.go

```diff
@@ -86,24 +85,25 @@ var (
 	errEventNotFoundInBatch = serviceerror.NewInternal("History event not found within expected batch")
 )
 
-var _ eventsCache = (*eventsCacheImpl)(nil)
-
-func newEventsCache(shardCtx ShardContext) eventsCache {
-	config := shardCtx.GetConfig()
-	shardID := convert.Int32Ptr(shardCtx.GetShardID())
-	return newEventsCacheWithOptions(config.EventsCacheInitialSize(), config.EventsCacheMaxSize(), config.EventsCacheTTL(),
-		shardCtx.GetHistoryManager(), false, shardCtx.GetLogger(), shardCtx.GetMetricsClient(), shardID)
-}
-
-func newEventsCacheWithOptions(initialSize, maxSize int, ttl time.Duration,
-	eventsV2Mgr persistence.HistoryManager, disabled bool, logger log.Logger, metrics metrics.Client, shardID *int32) *eventsCacheImpl {
+var _ Cache = (*CacheImpl)(nil)
+
+func NewEventsCache(
+	shardID *int32,
```

Comment:
Why *int32? Shouldn't it be int32?

Reply:
just moving things around, keeping the logic as is